tests/bluechi_test/bluechilib/bluechictl.py

```diff
@@ -0,0 +1,141 @@
+
+
+class EventEvaluator:
+    def process_line(self, line: str) -> None:
+        pass
+
+    def processing_finished(self) -> bool:
+        pass
+
+
+class SimpleEventEvaluator(EventEvaluator):
+    def __init__(self, expected_patterns: Set[Pattern]):
+        self.expected_patterns = copy.copy(expected_patterns)
+
+    def process_line(self, line: str) -> None:
+        for pattern in copy.copy(self.expected_patterns):
+            if pattern.match(line):
+                print(f"Matched line '{line}'")
+                self.expected_patterns.remove(pattern)
+                break
+
+    def processing_finished(self) -> bool:
+        return len(self.expected_patterns) == 0
+
+
+class BluechiCtlBackgroundRunner:
+    def __init__(self, command: List, evaluator: EventEvaluator):
+        self.command = command
+        self.evaluator = evaluator
+        self.thread = threading.Thread(target=self.process_events)
+        self.failsafe_thread = threading.Thread(target=self.timeout_guard, daemon=True)
+        self.finished = False
+        self.bluechictl_proc = None
+
+    def process_events(self):
+        print("Starting process_events")
+        with tempfile.NamedTemporaryFile() as out_file:
+            try:
+                self.bluechictl_proc = subprocess.Popen(self.command, stdout=out_file, bufsize=1)
+
+                with FileFollower(out_file.name) as bluechictl_out:
+                    while self.bluechictl_proc.poll() is None and not self.finished:
+                        for line in bluechictl_out:
+                            print(f"Evaluating line '{line}'")
+                            self.evaluator.process_line(line)
+                            if self.evaluator.processing_finished():
+                                print("BluechiCtl output evaluation successfully finished")
+                                self.finished = True
+                                break
+
+                        # Wait for the new output from bluechictl monitor
+                        time.sleep(0.5)
+            finally:
+                print("Finalizing process")
+                if self.bluechictl_proc:
+                    self.bluechictl_proc.send_signal(signal.SIGINT)
+                    self.bluechictl_proc.wait()
+                    self.bluechictl_proc = None
+
+    def timeout_guard(self):
+        time.sleep(10)
+        print(f"Waiting for expected output in {self.command} has timed out")
+        self.bluechictl_proc.send_signal(signal.SIGINT)
+        self.bluechictl_proc.wait()
+
+    def start(self):
+        print("Starting thread...")
+        self.thread.start()
+        self.failsafe_thread.start()
+
+    def stop(self):
+        self.finished = True
+        self.thread.join()
+
+    def found_all_patterns(self):
+        return self.evaluator.processing_finished()
+
+
+class BluechiCtl:
+    def run(self, args: List) -> Tuple[str, str]:
+        command = ["bluechictl"] + args
```

Comment:
```suggestion
        command = ["/usr/bin/bluechictl"] + args
```
We can probably reintroduce the path here. It was an error on my side removing it.

Reply:
Forgot to roll back this change